userspace/libsinsp/metrics_collector.cpp

```diff
@@ -0,0 +1,559 @@
+			pss = convert_memory(pss_unit, METRIC_VALUE_UNIT_MEMORY_MEGABYTES, pss);
+			memory_used_host = convert_memory(memory_used_host_unit, METRIC_VALUE_UNIT_MEMORY_MEGABYTES, memory_used_host);
+			container_memory_usage = convert_memory(container_memory_usage_unit, METRIC_VALUE_UNIT_MEMORY_MEGABYTES, container_memory_usage);
+			rss_unit = vsz_unit = pss_unit = memory_used_host_unit = container_memory_usage_unit = METRIC_VALUE_UNIT_MEMORY_MEGABYTES;
+			rss_type = vsz_type = pss_type = memory_used_host_type = container_memory_usage_type = METRIC_VALUE_TYPE_D;
+		}
+		// Resource utilization of the agent itself
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_CPU_PERC],
+		METRICS_V2_RESOURCE_UTILIZATION, METRIC_VALUE_TYPE_D, METRIC_VALUE_UNIT_PERC, METRIC_VALUE_NON_MONOTONIC_CURRENT, cpu_usage_perc));
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_MEMORY_RSS],
+		METRICS_V2_RESOURCE_UTILIZATION, rss_type, rss_unit, METRIC_VALUE_NON_MONOTONIC_CURRENT, rss));
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_MEMORY_VSZ],
+		METRICS_V2_RESOURCE_UTILIZATION, vsz_type, vsz_unit, METRIC_VALUE_NON_MONOTONIC_CURRENT, vsz));
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_MEMORY_PSS],
+		METRICS_V2_RESOURCE_UTILIZATION, pss_type, pss_unit, METRIC_VALUE_NON_MONOTONIC_CURRENT, pss));
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_CONTAINER_MEMORY],
+		METRICS_V2_RESOURCE_UTILIZATION, container_memory_usage_type, container_memory_usage_unit, METRIC_VALUE_NON_MONOTONIC_CURRENT, container_memory_usage));
+		// Resource utilization / load indicators of the underlying host
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_CPU_PERC_TOTAL_HOST],
+		METRICS_V2_RESOURCE_UTILIZATION, METRIC_VALUE_TYPE_D, METRIC_VALUE_UNIT_PERC, METRIC_VALUE_NON_MONOTONIC_CURRENT, cpu_usage_perc_total_host));
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_PROCS_HOST],
+		METRICS_V2_RESOURCE_UTILIZATION, METRIC_VALUE_TYPE_U32, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_NON_MONOTONIC_CURRENT, procs_running_host));
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_MEMORY_TOTAL_HOST],
+		METRICS_V2_RESOURCE_UTILIZATION, memory_used_host_type, memory_used_host_unit, METRIC_VALUE_NON_MONOTONIC_CURRENT, memory_used_host));
+		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_FDS_TOTAL_HOST],
+		METRICS_V2_RESOURCE_UTILIZATION, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_NON_MONOTONIC_CURRENT, open_fds_host));
+	}
+
```

Comment:
This block is too compact, reading is harder than it should be.
```suggestion
		const scap_agent_info* agent_info = m_inspector->get_agent_info();
        uint64_t container_memory_usage = get_container_memory_usage();
		
		uint32_t rss{0}, vsz{0}, pss{0}, procs_running_host{0};
		uint64_t memory_used_host{0}, open_fds_host{0};
		double cpu_usage_perc{0.0}, cpu_usage_perc_total_host{0.0};
	
    	metrics_v2_value_unit rss_unit{METRIC_VALUE_UNIT_MEMORY_KILOBYTES},
                                vsz_unit{METRIC_VALUE_UNIT_MEMORY_KILOBYTES},
		                        pss_unit{METRIC_VALUE_UNIT_MEMORY_KILOBYTES},
                                memory_used_host_unit{METRIC_VALUE_UNIT_MEMORY_KILOBYTES},
                                container_memory_usage_unit{METRIC_VALUE_UNIT_MEMORY_BYTES};

		metrics_v2_value_type rss_type{METRIC_VALUE_TYPE_U32},
                                vsz_type{METRIC_VALUE_TYPE_U32},
		                        pss_type{METRIC_VALUE_TYPE_U32},
                                memory_used_host_type{METRIC_VALUE_TYPE_U32},
                                container_memory_usage_type{METRIC_VALUE_TYPE_U64};

		get_cpu_usage_and_total_procs(agent_info->start_time, cpu_usage_perc, cpu_usage_perc_total_host, procs_running_host);
		get_rss_vsz_pss_total_memory_and_open_fds(rss, vsz, pss, memory_used_host, open_fds_host);

		if(m_convert_memory_to_mb)
		{
			rss = convert_memory(rss_unit, METRIC_VALUE_UNIT_MEMORY_MEGABYTES, rss);
			vsz = convert_memory(vsz_unit, METRIC_VALUE_UNIT_MEMORY_MEGABYTES, vsz);
			pss = convert_memory(pss_unit, METRIC_VALUE_UNIT_MEMORY_MEGABYTES, pss);
			memory_used_host = convert_memory(memory_used_host_unit, METRIC_VALUE_UNIT_MEMORY_MEGABYTES, memory_used_host);
			container_memory_usage = convert_memory(container_memory_usage_unit, METRIC_VALUE_UNIT_MEMORY_MEGABYTES, container_memory_usage);
			rss_unit = vsz_unit = pss_unit = memory_used_host_unit = container_memory_usage_unit = METRIC_VALUE_UNIT_MEMORY_MEGABYTES;
			rss_type = vsz_type = pss_type = memory_used_host_type = container_memory_usage_type = METRIC_VALUE_TYPE_D;
		}

		// Resource utilization of the agent itself
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_CPU_PERC],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            METRIC_VALUE_TYPE_D,
                                            METRIC_VALUE_UNIT_PERC,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            cpu_usage_perc));
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_MEMORY_RSS],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            rss_type,
                                            rss_unit,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            rss));
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_MEMORY_VSZ],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            vsz_type,
                                            vsz_unit,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            vsz));
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_MEMORY_PSS],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            pss_type, 
                                            pss_unit,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            pss));
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_CONTAINER_MEMORY],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            container_memory_usage_type,
                                            container_memory_usage_unit,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            container_memory_usage));

		// Resource utilization / load indicators of the underlying host
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_CPU_PERC_TOTAL_HOST],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            METRIC_VALUE_TYPE_D,
                                            METRIC_VALUE_UNIT_PERC,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            cpu_usage_perc_total_host));
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_PROCS_HOST],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            METRIC_VALUE_TYPE_U32,
                                            METRIC_VALUE_UNIT_COUNT,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            procs_running_host));
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_MEMORY_TOTAL_HOST],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            memory_used_host_type,
                                            memory_used_host_unit,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            memory_used_host));
		m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_RESOURCE_UTILIZATION_FDS_TOTAL_HOST],
		                                    METRICS_V2_RESOURCE_UTILIZATION,
                                            METRIC_VALUE_TYPE_U64,
                                            METRIC_VALUE_UNIT_COUNT,
                                            METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                            open_fds_host));
	}

```

Reply:
The diff looks rather strange as it is not wide enough to properly show the end result.

Reply:
As stated in my new comment, I'll leave formatting corrections for the end. Thanks!